src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.NetStandard10.cs

```diff
@@ -0,0 +1,401 @@
+        //
+        private static object s_intrinsicNullableKey = new object();
+
+        private static object s_syncObject = new object();
+
+        /// <devdoc>
+        ///     Creates a new ReflectTypeDescriptionProvider.  The type is the
+        ///     type we will obtain type information for.
+        /// </devdoc>
+        internal ReflectTypeDescriptionProvider()
+        {
+        }
+
+        /// <devdoc> 
+        ///      This is a table we create for intrinsic types. 
+        ///      There should be entries here ONLY for intrinsic 
+        ///      types, as all other types we should be able to 
+        ///      add attributes directly as metadata. 
+        /// </devdoc> 
+        private static Dictionary<object, object> IntrinsicTypeConverters
+        {
+            get
+            {
+                Debug.Assert(Monitor.IsEntered(s_syncObject));
+
+                // It is not worth taking a lock for this -- worst case of a collision
+                // would build two tables, one that garbage collects very quickly.
+                //
+                if (ReflectTypeDescriptionProvider.s_intrinsicConverters == null)
+                {
+                    Dictionary<object, object> temp = new Dictionary<object, object>();
+
+                    // Add the intrinsics
+                    //
+                    temp[typeof(bool)] = typeof(BooleanConverter);
+                    temp[typeof(byte)] = typeof(ByteConverter);
+                    temp[typeof(SByte)] = typeof(SByteConverter);
+                    temp[typeof(char)] = typeof(CharConverter);
+                    temp[typeof(double)] = typeof(DoubleConverter);
+                    temp[typeof(string)] = typeof(StringConverter);
+                    temp[typeof(int)] = typeof(Int32Converter);
+                    temp[typeof(short)] = typeof(Int16Converter);
+                    temp[typeof(long)] = typeof(Int64Converter);
+                    temp[typeof(float)] = typeof(SingleConverter);
+                    temp[typeof(UInt16)] = typeof(UInt16Converter);
+                    temp[typeof(UInt32)] = typeof(UInt32Converter);
+                    temp[typeof(UInt64)] = typeof(UInt64Converter);
+                    temp[typeof(object)] = typeof(TypeConverter);
+                    temp[typeof(void)] = typeof(TypeConverter);
+                    temp[typeof(DateTime)] = typeof(DateTimeConverter);
+                    temp[typeof(DateTimeOffset)] = typeof(DateTimeOffsetConverter);
+                    temp[typeof(Decimal)] = typeof(DecimalConverter);
+                    temp[typeof(TimeSpan)] = typeof(TimeSpanConverter);
+                    temp[typeof(Guid)] = typeof(GuidConverter);
+                    temp[typeof(Array)] = typeof(ArrayConverter);
+                    temp[typeof(ICollection)] = typeof(CollectionConverter);
+                    temp[typeof(Enum)] = typeof(EnumConverter);
```

Comment:
Cool - I didn't know `UriTypeConverter` was included. I'll update our branch and get that in there.

As far as keeping it in sync, that's a good question. It'll take a bit of refactoring to only have one source. The ReflectTypeDescriptionProvider in master right now is a greatly simplified version of the full ReflectTypeDescriptionProvider and before I take the time to refactor it into one file, I want to ensure the concept of only exposing .NET Standard 1.0 surface area for wp8/win8 is the right thing to do.